### PR TITLE
feat: TreeView + VoicesNodeMenu

### LINKED ISSUE #733: Tree view

### PR DESCRIPTION
**Pull request by damian-molinski**
# Description

This is simple implementation of `TreeView` which supports only **one** level of depth.
Reasoning here is that google has their package [two_dimensional_scrollables](https://pub.dev/packages/two_dimensional_scrollables) which has `TreeView` implemented but at the moment it's bugged and we can not used it. It's also Scrollable so we need to specify height, which is not ideal. 

## Related Issue(s)

Closes #733 

## Screenshots

https://github.com/user-attachments/assets/145cfad6-2d4b-4806-a27b-2f6173843dc4

## Related Pull Requests

Part of #719, #729

## Please confirm the following checks

* [x] My code follows the style guidelines of this project
* [x] I have performed a self-review of my code
* [x] I have commented my code, particularly in hard-to-understand areas
* [x] I have made corresponding changes to the documentation
* [x] My changes generate no new warnings
* [x] I have added tests that prove my fix is effective or that my feature works
* [x] New and existing unit tests pass locally with my changes
* [x] Any dependent changes have been merged and published in downstream module

### CODE REVIEW

File: catalyst_voices/lib/widgets/common/simple_tree_view.dart [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+        child: ConstrainedBox(
+          constraints: const BoxConstraints.tightFor(height: 32),
+          child: Material(
+            type: MaterialType.transparency,
+            child: InkWell(
+              onTap: onTap,
+              child: Padding(
+                padding: const EdgeInsets.symmetric(horizontal: 16),
+                child: Row(
+                  children: <Widget>[
```
Comment (dtscalac) [inline catalyst_voices/lib/widgets/common/simple_tree_view.dart:73]: <Widget> type not needed, in general specifying the type of the list where list of widgets is expected was required very long time ago, noways we skip it... Are you having some issues with linter?

File: catalyst_voices/lib/widgets/common/simple_tree_view.dart [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+              child: Offstage(
+                offstage: !showBottom,
+                child: const VerticalDivider(),
+              ),
+            ),
+          ],
+        ),
+        const Row(
+          children: [
+            Expanded(child: SizedBox()),
```
Comment (dtscalac) [inline catalyst_voices/lib/widgets/common/simple_tree_view.dart:193]: Spacer() wouldn't be better?

File: catalyst_voices/lib/widgets/common/simple_tree_view.dart (hunk 1 above)
Comment (damian-molinski) [reply catalyst_voices/lib/widgets/common/simple_tree_view.dart:73]: actually you're right. In this case it's not needed, even with `separatedBy`

**Review by dtscalac — APPROVED** [review]
lgtm

**Pull request merged**